File: HtmlPrototype/game/gameplay/player.js

function player(playerNumber, playerName){
    var $playerNumber = playerNumber;
    var $playerName = playerName;

    var $resources = new playerResources();

    this.playerNumber = function(){
        return $playerNumber;
    };
    this.playerName = function(){
        return $playerName;
    };

    this.resources = function(){
        return $resources;
    };

    this.modifyResources = function(resourceModification){
        $resources.addFood(resourceModification.food())
        $resources.addOre(resourceModification.ore())
        $resources.addTech(resourceModification.tech())
    };

}

File: StarBastard/StarBastard.Core/Gameplay/Player.cs
namespace StarBastard.Core.Gameplay
{
    public class Player
    {
        public int PlayerNumber { get; set; }
        public string PlayerName { get; set; }
        public ResourceDelta Resources { get; set; }

        public Player(int playerNumber, string playerName)
        {
            PlayerNumber = playerNumber;
            PlayerName = playerName;
        }
        
        public void ModifyResources(ResourceDelta resourceModification)
        {
            Resources.AddFood(resourceModification.Food);
            Resources.AddOre(resourceModification.Ore);
            Resources.AddTech(resourceModification.Tech);

        }
    }
}